aten/src/ATen/native/cpu/CrossKernal.cpp

```diff
@@ -0,0 +1,72 @@
+#include <ATen/native/Cross.h>
+
+#include <numeric>
+#include <iterator>
+#include <algorithm>
+
+#include <ATen/Dispatch.h>
+#include <ATen/Parallel.h>
+#include <ATen/cpu/vml.h>
+namespace at { namespace native { namespace {
+
+template<typename scalar_t>
+static void apply_cross(Tensor& result, const Tensor& a, const Tensor& b, const int64_t dim) {
+  int64_t total = a.numel() / 3;
+  int64_t a_stride = a.stride(dim);
+  int64_t b_stride = b.stride(dim);
+  int64_t r_stride = result.stride(dim);
+
+  scalar_t *a_ptr = a.data<scalar_t>();
+  scalar_t *b_ptr = b.data<scalar_t>();
+  scalar_t *r_ptr = result.data<scalar_t>();
+
+  parallel_for(0, total, internal::GRAIN_SIZE, [&](int64_t s, int64_t e) {
+    int64_t counter[a.dim()];
+    int64_t v = s;
+    int64_t start = 0;
+    for (int64_t i = 0; i < a.dim(); i++) {
+      if (i == dim) continue;
+      counter[i] = v % a.size(i);
```

Comment:
can you use more descriptive variable names than `counter`, `v`, etc?

Reply:
Fixed